python/pyarrow/tests/test_array.py

```diff
@@ -536,6 +536,26 @@ def test_array_slice_negative_step():
         assert result.equals(expected)
 
 
+def test_arange():
+    cases = [
+        (0, 10),  # Default step
+        (0, 10, 1),
+        (0, 10, 2),
+        (10, 0, -1),
+        (10, 0, -2),
+        (0, 0),       # Empty array
+        (0, 10, -1),  # Empty array
+        (10, 0, 1),   # Empty array
+    ]
+    for case in cases:
+        result = pa.arange(*case)
```

Comment:
Validating results is a good habit to take in unit tests
```suggestion
        result = pa.arange(*case)
        result.validate(full=True)
```